pkg/machineproviders/providers/machineproviders.go

```diff
@@ -50,3 +52,19 @@ func NewMachineProvider(ctx context.Context, logger logr.Logger, cl client.Clien
 		return nil, fmt.Errorf("%w: %s", errUnexpectedMachineType, cpms.Spec.Template.MachineType)
 	}
 }
+
+// GetMachinesTypeMeta returns proper TypeMeta from ControlPlaneMachineSetMachineType.
+func GetMachinesTypeMeta(cpmsMachinesType machinev1.ControlPlaneMachineSetMachineType) (metav1.TypeMeta, error) {
```

Comment:
Nit, I'd remove the S here
```suggestion
// GetMachineTypeMeta returns proper TypeMeta from ControlPlaneMachineSetMachineType.
func GetMachineTypeMeta(cpmsMachinesType machinev1.ControlPlaneMachineSetMachineType) (metav1.TypeMeta, error) {
```